lib/package_manager.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+
+function package_manager::determine_package_manager() {
+	local build_dir="${1}"
+	local package_managers_found=()
```

Comment:
Based on the logic below, an indexed array will work fine but I do think you should consider an associative array here instead. Mostly as a way to guarantee you only have one unique entry per package manager detected to prevent a small change in logic introducing a subtle bug where you could unintentionally end up with a double entry (e.g.; `package_managers_found=(pip pip)`)

Reply:
It also seems that with associative arrays:
- it's not possible to fetch a single element (only "display all keys")
- the keys when listed are not listed in insertion order, which would break the package manager order of precedence requirement of the current implementation